pkg/services/ngalert/notifier/receivers.go

```diff
@@ -197,6 +199,39 @@ func (am *Alertmanager) TestReceivers(ctx context.Context, c apimodels.TestRecei
 	return newTestReceiversResult(testAlert, append(invalid, results...), now), nil
 }
 
+func (am *Alertmanager) GetReceivers(ctx context.Context) apimodels.Receivers {
+	var receivers []*notify.Receiver
+	receivers = append(receivers, am.receivers...)
+	var apiReceivers apimodels.Receivers
+	for _, rcv := range receivers {
+		// Build integrations slice
+		var apiIntegrations []*models.Integration
+		for _, integration := range rcv.Integrations() {
+			lastNotify, lastNotifyDuration, lastNotifyError := integration.GetReport()
+			apiIntegrations = append(apiIntegrations, &models.Integration{
+				LastNotifyDuration: lastNotifyDuration.String(),
+				LastNotify:         strfmt.DateTime(lastNotify),
+				LastError: func() string {
```

Comment:
Shouldn't `LastError` be `error` instead of `string`?

Reply:
This is the model that is going to be sent back to the client, that's why we're using a string.
Maybe this conversion to `models.Integration` should be done in the API layer? (`api_alertmanager.go`)

Reply:
I think what's weird here is that we are returning a struct for JSON responses (`"github.com/prometheus/alertmanager/api/v2/models"`) from code that does not interface with HTTP APIs.

What I mean is that the Alertmanager does not receive and return JSON requests and responses, but that's what the structs in `"github.com/prometheus/alertmanager/api/v2/models"` are. The erasure of the error (both the error type and wrapped errors) also means that the error is opaque to users of this function other than the HTTP API.

You could return a struct that has `lastError error` which is not defined in `"github.com/prometheus/alertmanager/api/v2/models"` but perhaps `"github.com/grafana/prometheus-alertmanager/notify"` and then just before sending the API response turn it into an API response.

What are your thoughts on this?

@alexweav @JacobsonMT @yuri-tceretian

Reply:
Makes sense! I'll try to address this today and find a better solution.

Reply:
If that's the case I think we can keep it as is, and if we want to change it we can do later!